pkg/datastore/datastore.go

```diff
@@ -342,6 +342,14 @@ type WatchOptions struct {
 	// If given the zero value, the datastore's default will be used. If smaller
 	// than the datastore's minimum, the minimum will be used.
 	CheckpointInterval time.Duration
+
+	// WatchBufferLength is the length of the buffer for the watch channel. If
+	// given the zero value, the datastore's default will be used.
+	WatchBufferLength uint16
```

Comment:
@josephschorr we probably want to specify some sort of upper limit in the various `Watch` implementations to make sure we don't OOMKill due to a massively large channel

Reply:
That's why its a uint16 :)